crates/miden-tx/src/executor/notes_checker.rs

```diff
@@ -51,56 +45,51 @@ where
         input_notes: InputNotes<InputNote>,
         tx_args: TransactionArgs,
         source_manager: Arc<dyn SourceManager>,
-    ) -> Result<NoteAccountExecution, TransactionExecutorError> {
-        // Check input notes
-        // ----------------------------------------------------------------------------------------
-
-        let mut successful_notes = vec![];
+    ) -> Result<(), Box<NoteConsumptionError>> {
```

Comment:
I'm not sure about returning "nothing" in case of success here. The way I think about this method is that its purpose is to classify a set of notes into successful and failed. In a way, even if all notes fail execution it is not really an error - we've successfully classified them.

So, maybe the return type should be similar to what it was before:
```rust
Result<NoteConsumptionInfo, TransactionExecutorError>
```
But `NoteConsumptionInfo` would now contain `Note` objects rather than just IDs - e.g., it could look like so:
```rust
pub struct NoteConsumptionInfo {
    pub successful: Vec<Note>, 
    pub failed: Vec<Note>,
}
```
I've removed `error` from here because it is not clear to me if we need to return this info as there isn't much we can do with it.

Reply:
Updated, thanks.